frontend/src/components/ExplorerCard/ExplorerCard.tsx

```diff
@@ -0,0 +1,251 @@
+	} = useGetAllViews(sourcepage);
+
+	useErrorNotification(error);
+
+	const handlePopOverClose = (): void => {
+		setIsOpen(false);
+	};
+
+	const handleOpenChange = (newOpen: boolean): void => {
+		setIsOpen(newOpen);
+	};
+
+	const viewName =
+		useGetSearchQueryParam(querySearchParams.viewName) || 'Query Builder';
+
+	const viewKey = useGetSearchQueryParam(querySearchParams.viewKey) || '';
+
+	const { mutateAsync: updateViewAsync } = useUpdateView({
+		compositeQuery: mapCompositeQueryFromQuery(currentQuery, panelType),
+		viewKey,
+		extraData: '',
+		sourcePage: sourcepage,
+		viewName,
+	});
+
+	const onUpdateQueryHandler = useCallback(async () => {
+		await updateQueryHandler({
+			compositeQuery: mapCompositeQueryFromQuery(currentQuery, panelType),
+			viewKey,
+			extraData: '',
+			sourcePage: sourcepage,
+			viewName,
+			notifications,
+			setIsQueryUpdated,
+			updateViewAsync,
+		});
+		refetchAllView();
+	}, [
+		currentQuery,
+		notifications,
+		panelType,
+		refetchAllView,
+		sourcepage,
+		updateViewAsync,
+		viewKey,
+		viewName,
+	]);
+
+	const onMenuItemSelectHandler = useCallback(
+		({ key }: { key: string }): void => {
+			const currentViewDetails = getViewDetailsUsingViewKey(key, data?.data?.data);
+			if (!currentViewDetails) return;
+			const { query, name, uuid } = currentViewDetails;
+
+			redirectWithQueryBuilderData(query, {
+				[querySearchParams.viewName]: name,
+				[querySearchParams.viewKey]: uuid,
+			});
+		},
+		[data?.data?.data, redirectWithQueryBuilderData],
+	);
+
+	useEffect(() => {
+		if (copyUrl.value) {
+			notifications.success({
+				message: 'Copied to clipboard',
+			});
+		}
+	}, [copyUrl.value, notifications]);
+
+	useEffect(() => {
+		setIsQueryUpdated(
+			isQueryUpdatedInView({
+				data: data?.data?.data,
+				stagedQuery,
+				viewKey,
+			}),
+		);
+	}, [
+		currentQuery,
+		data?.data?.data,
+		stagedQuery,
+		stagedQuery?.builder.queryData,
+		viewKey,
+	]);
+
+	const generatorMenuItems = useCallback(
+		(data: ViewProps[] | undefined): MenuProps['items'] => {
+			if (!data) return [];
+
+			return data.map((view) => ({
+				key: view.uuid,
+				label: (
+					<MenuItemGenerator
+						viewName={view.name}
+						viewKey={viewKey}
+						createdBy={view.createdBy}
+						uuid={view.uuid}
+						refetchAllView={refetchAllView}
+						onMenuItemSelectHandler={onMenuItemSelectHandler}
+					/>
+				),
+			}));
+		},
+		[onMenuItemSelectHandler, refetchAllView, viewKey],
+	);
+
+	const updateMenuList = useMemo(() => generatorMenuItems(data?.data.data), [
+		data?.data.data,
+		generatorMenuItems,
+	]);
+
+	const menu = useMemo(
+		(): MenuProps => ({
+			items: updateMenuList,
+		}),
+		[updateMenuList],
+	);
+
+	return (
+		<>
+			<ExplorerCardHeadContainer size="small">
+				<Row>
+					<Col span={6}>
+						<Space>
+							<Typography>{viewName}</Typography>
+							<TextToolTip
+								url={ExploreHeaderToolTip.url}
+								text={ExploreHeaderToolTip.text}
```

Comment:
Can we default to some value in case ExploreHeaderToolTip.text is undefined ?

Reply:
So, here we are getting `ExploreHeaderToolTip` from config, which is hard coded in the code. So it will never be `undefined`, we will always get a value.